gov/politeia/proposals.go

```diff
@@ -188,26 +207,61 @@ func (db *ProposalDB) saveProposals(URLParams string) (int, error) {
 
 		err = db.dbP.Save(val)
 
-		// In the rare case scenario that the current proposal has a duplicate refID
-		// append an integer value to it till it becomes unique.
+		// When a duplicate CensorshipRecord struct is detected this "already exists"
+		// error is thrown, this means that some edits were made to an older version
+		// of the proposal and its fixed by updating the new changes. In another
+		// case that is more rare, is that the current proposal could have a Name
+		// that generates a RefID similar to one already in the db and appending
+		// integers to it till it becomes unique is the solution.
 		if err == storm.ErrAlreadyExists {
-			c := 1
-			for {
-				val.RefID += strconv.Itoa(c)
-				err = db.dbP.Save(val)
-				if err == nil || err != storm.ErrAlreadyExists {
-					break
+			// Check if the proposal token already exists in the db.
+			data, newErr := db.proposal("TokenVal", val.TokenVal)
+			if newErr == nil && data != nil {
+				// The proposal token already exists thus trigger an update with
+				// the latest details.
+				valCopy := *val
+				valCopy.ID = data.ID
+				prefixStr := ""
+
+				for k := 1; k <= maxLoop; k++ {
+					valCopy.RefID += prefixStr
+					// Attempt to update the old entry.
+					newErr = db.dbP.Update(&valCopy)
+					if newErr == nil || newErr != storm.ErrAlreadyExists {
```

Comment:
I don't understand this condition. First, `newErr != storm.ErrAlreadyExists` is covered by `newErr == nil`.  Second, what about if `newErr` is some other fatal error?

Reply:
The loop and if statement are attempting to solve `storm.ErrAlreadyExists` error thus the loop should stop once an error other than `storm.ErrAlreadyExists` and `nil` is returned.
There exists two error variables `err` and `newErr` where `err` is the main error to be considered if the attempted update/save new data fails. 

Its expected that by the time the loop above exits either of the following will have happened;
1. Update wasn't successful if all the suffixes added by the fifth (`maxLoop`) did not make it unique.
2. Another error other than `storm.ErrAlreadyExists` was returned thus update failed.
3. Update was successful and `newErr` is finally equal to `nil`.

Reply:
Actually `newErr` variable is not necessary.

Reply:
I have dropped the `newErr` error variable as it wasn't necessary.